lib/database/user.php

```diff
@@ -1374,10 +1374,12 @@ function getUserListByPerms($sortBy, $offset, $count, &$dataOut, $requestedBy, &
     $permsFilter = null;
 
     settype($perms, 'integer');
-    if ($perms >= Permissions::Spam && $perms <= Permissions::Unregistered || $perms == Permissions::SuperUser) {
+    if ($perms >= Permissions::Spam && $perms <= Permissions::Unregistered) {
         $permsFilter = "ua.Permissions = $perms ";
-    } elseif ($perms >= Permissions::Registered && $perms <= Permissions::Admin) {
+    } elseif ($perms >= Permissions::Registered) {
         $permsFilter = "ua.Permissions >= $perms ";
+    } elseif ($perms > Permissions::Registered && $perms <= Permissions::Root) {
+        $permsFilter = "ua.Permissions == $perms ";
```

Comment:
this won't work. line 1379 ensures it never reaches 1381, and there might be an issue with "==" vs "=" in the query.

this is more what I was thinking:
```
if ($perms == Permissions::Registered) {
    // everything Registered or above is registered
    $permsFilter = "ua.Permissions >= $perms ";
} else {
    // specific type requested, only show that.
    $permsFilter = "ua.Permissions = $perms ";
}
```
Although, Developer should probably be >=  as well, so admins show up as developers.